src/node_kernels/SDRSSTBLTM2015NodeKernel.C

```diff
@@ -0,0 +1,184 @@
+  tkeProdLimitRatio_ = realm.get_turb_model_constant(TM_tkeProdLimitRatio);
+  sigmaWTwo_ = realm.get_turb_model_constant(TM_sigmaWTwo);
+  betaOne_ = realm.get_turb_model_constant(TM_betaOne);
+  betaTwo_ = realm.get_turb_model_constant(TM_betaTwo);
+  gammaOne_ = realm.get_turb_model_constant(TM_gammaOne);
+  gammaTwo_ = realm.get_turb_model_constant(TM_gammaTwo);
+  sdrAmb_ = realm.get_turb_model_constant(TM_sdrAmb);
+
+  lengthScaleLimiter_ = realm.solutionOptions_->lengthScaleLimiter_;
+  if (lengthScaleLimiter_) {
+    const NodeKernelTraits::DblType earthAngularVelocity =
+      realm.solutionOptions_->earthAngularVelocity_;
+    const NodeKernelTraits::DblType pi = std::acos(-1.0);
+    const NodeKernelTraits::DblType latitude =
+      realm.solutionOptions_->latitude_ * pi / 180.0;
+    corfac_ = 2.0 * earthAngularVelocity * std::sin(latitude);
+    referenceVelocity_ = realm.solutionOptions_->referenceVelocity_;
+  }
+}
+
+KOKKOS_FUNCTION
+void
+SDRSSTBLTM2015NodeKernel::execute(
+  NodeKernelTraits::LhsType& lhs,
+  NodeKernelTraits::RhsType& rhs,
+  const stk::mesh::FastMeshIndex& node)
+{
+  using DblType = NodeKernelTraits::DblType;
+
+  const DblType tke = tke_.get(node, 0);
+  const DblType sdr = sdr_.get(node, 0);
+  const DblType density = density_.get(node, 0);
+  const DblType tvisc = tvisc_.get(node, 0);
+  const DblType dVol = dualNodalVolume_.get(node, 0);
+  const DblType fOneBlend = fOneBlend_.get(node, 0);
+
+  DblType Pk = 0.0;
+  DblType crossDiff = 0.0;
+//================= original model w/o transition ================//
+//  for (int i = 0; i < nDim_; ++i) {
+//    crossDiff += dkdx_.get(node, i) * dwdx_.get(node, i);
+//    const int offset = nDim_ * i;
+//    for (int j = 0; j < nDim_; ++j) {
+//      const auto dudxij = dudx_.get(node, offset + j);
+//      Pk += dudxij * (dudxij + dudx_.get(node, j * nDim_ + i));
+//    }
+//  }
+//  Pk *= tvisc;
+//
+//  const DblType Dk = betaStar_ * density * sdr * tke;
+//
+//  // Clip production term and clip negative productions
+//  Pk = stk::math::min(tkeProdLimitRatio_ * Dk, stk::math::max(Pk, 0.0));
+//
+//==================Transition model============================//
+  DblType sijMag    = 0.0;
+  DblType vortMag   = 0.0;
+
+  for (int i=0; i < nDim_; ++i) {
+    crossDiff += dkdx_.get(node, i) * dwdx_.get(node, i);
+    const int offset = nDim_ * i;
+    for (int j=0; j < nDim_; ++j) {
+     const double duidxj = dudx_.get(node, nDim_ * i + j);
+     const double dujdxi = dudx_.get(node, nDim_ * j + i);
+
+     const double rateOfStrain = 0.5 * (duidxj + dujdxi);
+     const double vortTensor = 0.5 * (duidxj - dujdxi);
+     sijMag += rateOfStrain * rateOfStrain;
+     vortMag += vortTensor * vortTensor;
+    }
+  }
+  sijMag = stk::math::sqrt(2.0*sijMag);
+  vortMag = stk::math::sqrt(2.0*vortMag);
+
+  Pk = tvisc * sijMag * vortMag; // Pk based on Kato-Launder formulation.
+
+//===============================================================//
```

Comment:
remove these types of comments throughout

Reply:
removed